### PR TITLE
Implements account sentinel

### PR DESCRIPTION
**Pull request by inetic**
Hope I didn't miss anything (possible though)

[Review on Reviewable](https://reviewable.io/reviews/maidsafe/sentinel/23)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @benjaminbollen

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/account_sentinel.rs [hunk 1]
```diff
@@ -15,244 +15,56 @@
...
-    fn get_signing_keys(&self, source: Name);
-}
+#[allow(dead_code)]
+pub struct AccountSentinel<Request, Name>
+    where Request: Eq + PartialOrd + Ord + Clone,
+          Name:    Eq + PartialOrd + Ord + Clone, {
 
-pub trait Source<Name> where Name: Eq + PartialOrd + Ord  + Clone {
-    fn get_source(&self) -> Name;
+    requests: LruCache<Request, Map<SerialisedClaim, Set<Name>>>,
```
Comment (benjaminbollen) [inline src/account_sentinel.rs]: Can we have an orderable claim? or am I missing something?

File: src/account_sentinel.rs [hunk 2]
```diff
@@ -15,244 +15,57 @@
...
-        //     return None;
-        // }
-
-        None
+    fn pick_median(map: &Map<Name, Claim>) -> Option<Claim> {
+        if map.is_empty() { return None }
+        let mut claims = map.iter().map(|(_, ref claim)| claim.clone())
+                            .collect::<Vec<_>>();
+        claims.sort();
+        Some(claims[claims.len() / 2].clone())
```
Comment (benjaminbollen) [inline src/account_sentinel.rs:69]: This is an integer division that always takes the floor of the rational result?

**Comment by benjaminbollen** [comment]
Looks good to me

File: src/account_sentinel.rs (hunk 2 above)
Comment (inetic) [inline src/account_sentinel.rs:69]: I do believe so: http://doc.rust-lang.org/0.11.0/num/integer/trait.Integer.html#tymethod.div_floor

**Pull request merged**